src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -21,5 +58,16 @@
             @await RenderSectionAsync("Footer", required: false)
         </footer>
     </div>
+    <script>
+        function setLocalizationCookie() {
+            var culturesList = document.getElementById('culturesList');
+            var culture = culturesList.options[culturesList.selectedIndex].value
+            var queryString = window.location.search;
+            var searchParams = new URLSearchParams(queryString);
+
+            searchParams.set("ui-culture", culture);
```

Comment:
@jtkech the queryString becomes much simpler with `URLSearchParams` ;)

Thanks